### PR TITLE
[14.0][MIG] cooperator: migration to 14.0

### PR DESCRIPTION
**Pull request by huguesdk**
## description

migrate cooperator to odoo 14.0

## odoo task

[t8755](https://gestion.coopiteasy.be/web#id=8755&model=project.task&view_type=form)

### CODE REVIEW

**Comment by carmenbianca** [comment]
You can disable eslint. Related to #227

**Comment by carmenbianca** [comment]
Did a little bit of tinkering on this to get the tests to run.

The removal of dependencies `email_template_config` and `partner_contact_address` merits discussion at CC, so I've added it to the agenda.

@victor-champonnois I think you're working on this this week. I believe the following must be done:

1. Repair failing tests.
2. Do a tiny little bit of functional testing. If some things don't work, consider writing automated tests for them.
3. Rebase other v14 PRs on top of this branch, and set `14.0-mig-cooperator` as the target branch of those PRs. Their tests should begin running, and you should be able to do functional testing on those branches.
4. Request code reviews once the tests are green (or almost green). Also poke the analysts to see how/if they want to do functional testing.

**Comment by carmenbianca** [comment]
`partner_contact_address` is a functional requirement of `cooperator`. Quoting Cath:

> This is useful => when you take a share as a company, you legally have to enter a representative. 
>
> I've put some video of this on the cooperator task
>
> [...]
>
> we need the representative on cooperator, that"s all I know

**Comment by victor-champonnois** [comment]
Adding a field in partner form, following the discussion on [this task](https://gestion.coopiteasy.be/web#id=8756&model=project.task&view_type=form&menu_id=) (@carmenbianca before you force push ^^⁾

File: cooperator/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+            sub_reg_num = sequence_id.next_by_id()
+            vals = {
+                "member": True,
+                "old_member": False,
+                "cooperator_register_number": int(sub_reg_num),
+            }
+        elif self.partner_id.old_member:
+            vals = {"member": True, "old_member": False}
+
+        return vals
```
Comment (carmenbianca) [inline cooperator/models/account_move.py:109]: This seems incredibly error-prone to me.

1. Method is named `get_*`, but in fact changes the state of the sequence register.
2. No explicit check whether `cooperator_register_number` already has a value. The check is implicit through the value of `old_member`.

Is there a better place to set `cooperator_register_number`?

File: cooperator/models/operation_request.py [hunk 2]
```diff
@@ -114,40 +113,34 @@ def _compute_subscription_amount(self):
         required=True,
         change_default=True,
         readonly=True,
-        default=lambda self: self.env["res.company"]._company_default_get(),
+        default=lambda self: self.env.company,
     )
 
-    invoice = fields.Many2one("account.invoice", string="Invoice")
+    invoice = fields.Many2one("account.move", string="Invoice")
```
Comment (carmenbianca) [inline cooperator/models/operation_request.py:119]: Good opportunity to rename to `invoice_id` maybe. Or `invoice_move_id`? Not sure.

**Review by carmenbianca — APPROVED** [review]
one comment, but LGTM.

massive work. thanks @huguesdk !

File: cooperator/models/operation_request.py (hunk 2 above)
Comment (huguesdk) [reply cooperator/models/operation_request.py:119]: indeed, it should be renamed. there are several other things that need renaming, and my understanding was that we will first do all of these in 12.0 as part of komunigi and then port them to 14.0. not doing it as part of the migration allows to reduce the probability of new bugs not directly related to the migration.

**Pull request merged**